### ISSUE TITLE
Custom field name support

### ISSUE THREAD

**Issue by gppam** [DarshanGowda0/GeoFlutterFire#120]
The field values 'geohash' and 'geopoint' is fine but wouldn't it be good to have a custom field name? Consider a large set of data with its geohash field name only labeled as 'g', it would be bad to retroactively change all of it to 'geohash' at this point. Or is it not allow-able?

**Comment by SonXuanNguyen2k** [comment]
@gppam I totally agree, I think the library maintainers should follow the practices recommended in the primary doc. The geographical point data type built into Firestore should not be used as [mentioned by the Firebase team](https://firebase.google.com/docs/firestore/manage-data/data-types):

> "At this time we do not recommend using this data type due to querying limitations. It is generally better to store latitude and longitude as separate numeric fields"
